src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3Connection.cs

```diff
@@ -456,116 +484,133 @@ private async Task AcceptStreamsAsync()
-                        throw HttpProtocolException.CreateHttp3ConnectionException(Http3ErrorCode.StreamCreationError);
-                    }
-
-                    buffer = new ArrayBuffer(initialSize: 32, usePool: true);
+                buffer = new ArrayBuffer(initialSize: 32, usePool: true);
+                int bytesRead;
 
-                    int bytesRead;
+                try
+                {
+                    // webtransport has a header of 3 bytes and in order to avoid reading
+                    // user content for webtransport stream we need at first to read 3 bytes
+                    byte[] header = new byte[3];
+                    bytesRead = await stream.ReadAsync(header, CancellationToken.None).ConfigureAwait(false);
+                    header.CopyTo(buffer.AvailableMemory);
+                }
+                catch (QuicException ex) when (ex.QuicError == QuicError.StreamAborted)
+                {
+                    // Treat identical to receiving 0. See below comment.
+                    bytesRead = 0;
+                }
 
-                    try
-                    {
-                        bytesRead = await stream.ReadAsync(buffer.AvailableMemory, CancellationToken.None).ConfigureAwait(false);
-                    }
-                    catch (QuicException ex) when (ex.QuicError == QuicError.StreamAborted)
-                    {
-                        // Treat identical to receiving 0. See below comment.
-                        bytesRead = 0;
-                    }
+                if (bytesRead == 0)
+                {
+                    // https://quicwg.org/base-drafts/draft-ietf-quic-http.html#name-unidirectional-streams
+                    // A sender can close or reset a unidirectional stream unless otherwise specified. A receiver MUST
+                    // tolerate unidirectional streams being closed or reset prior to the reception of the unidirectional
+                    // stream header.
+                    return;
+                }
 
-                    if (bytesRead == 0)
-                    {
-                        // https://quicwg.org/base-drafts/draft-ietf-quic-http.html#name-unidirectional-streams
-                        // A sender can close or reset a unidirectional stream unless otherwise specified. A receiver MUST
-                        // tolerate unidirectional streams being closed or reset prior to the reception of the unidirectional
-                        // stream header.
-                        return;
-                    }
+                buffer.Commit(bytesRead);
+                bool foundStreamType = VariableLengthIntegerHelper.TryRead(buffer.ActiveSpan, out long streamType, out bytesRead);
+                while (foundStreamType is false)
+                {
+                    foundStreamType = VariableLengthIntegerHelper.TryRead(buffer.AvailableSpan, out streamType, out bytesRead);
```

Comment:
What do you think is the difference between `buffer.ActiveSpan` and `buffer.AvailableSpan`? What do you think will be read here?

Reply:
Indeed I mixed 2 actions which I needed to do:
Firstly I needed to read more from the stream inside the buffer, and afterwards I had to read from the buffer again the integer
Instead I tried to read from a not written space in the buffer *palmface*
Fixed now